### PR TITLE
Create basic skeleton of Flutter project

### PR DESCRIPTION
**Pull request by Levi-Lesches**
Committing the basics of running `flutter create` but also a handful of already-documented library files that will be filled in as we go in `lib`. 

Also updates the README and adds a few helpful GitHub Actions workflows: 
- `Analyzer`, which runs `flutter analyze` on pull requests
- `Documentation`, which runs `dartdoc` on new commits to `main`
  - the documentation is sent to the `docs` directory in the `documentation` branch
- My own [Blocking Issues](https://github.com/Levi-Lesches/blocking-issues) bot, which can mark certain issues or pull requests as dependent on others, and pull requests won't mergeable until the blocking pull request/issue is merged/closed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,3 +2,66 @@
...
+Every folder in this project has a purpose, but you’ll mostly need the `lib` folder. 
+
+- `.github`: Contains GitHub specific files, like Actions workflow files
+- `lib`: Contains all the cross-platform Flutter code for the front and back ends
+- `linux`: Contains Linux-specific code generated by `flutter create`
+- `macos`: Contains Mac-specific code generated by `flutter create`
+- `test`: Contains unit and integration tests for the project
+- `windows`: Contains Windows-specific code generated by `flutter create`
+
+It’s worth noting that the platform-specific directories **do not** contain the compiled executables. Those would be under the `build` directory, which is in the `.gitignore` file. These directories simply hold platform-specific configuration that was automatically generated by the `flutter create` command an may have since been customized by us.
```
Comment (superdan-t) [inline README.md:17]: > command an may have since been customized by us.

Small Typo?

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+name: Documentation
+
+# Documents the Dart code within this project with the `dartdoc` tool.
+# Publishes the result to a separate branch, `documentation`, under the `docs` folder. 
+# Set GitHub pages to publish that target to get an online documentation site. 
+
+on:
+  push:
+    branches: [ master ]
```
Comment (superdan-t) [inline .github/workflows/documentation.yml:9]: I'm no expert in GitHub workflows but this file uses the branch name `master` in multiple places. Should it be changed to `main`?

**Review by superdan-t — APPROVED** [review]
I left two comments, which may require very minor revisions.

I'm marking this as approved since addressing them should be trivial

File: .github/workflows/documentation.yml (hunk 2 above)
Comment (Levi-Lesches) [reply .github/workflows/documentation.yml:9]: Oh yes, thanks, this is from my old repo before `main` was introduced.

**Pull request merged**